lib/cocoapods-core/github.rb

```diff
@@ -96,10 +96,14 @@ def self.modified_since_commit(url, commit)
         'Accept' => 'application/vnd.github.chitauri-preview+sha',
         'If-None-Match' => %("#{commit}"),
       }
-      response = REST.get(request_url, headers)
-      code = response.status_code
 
-      code != 304
+      begin
+        response = REST.get(request_url, headers)
+        code = response.status_code
+        code != 304
+      rescue
+        raise Informative, "Failed to check Specs repository: #{url}"
```

Comment:
Perhaps?

`"Failed to check Specs repository: #{url}"` -> `"Failed to access the Specs Repository: #{url} - Please check if you are offline, or that GitHub is down.`